contracts/Governance.sol

```diff
@@ -0,0 +1,285 @@
+        proposals[_id].uriArguments = _uriArguments;
+
+        proposals[_id].descriptionHash = keccak256(proposals[_id].uriDescription);
+        proposals[_id].argumentsHash = keccak256(proposals[_id].uriArguments);
+
+        emit ProposalCreated(_id, _destination, _amount, _data, _uriDescription, proposals[_id].descriptionHash, _uriArguments, proposals[_id].argumentsHash);
+    }
+
+
+    /** @dev Request registering a proposal to the proposal list
+     *  @param _id ID of a proposal
+     */
+    function requestRegisteringProposal(bytes32 _id) public payable onlyWhenProposalIsNew(_id) {
+        proposalList.requestRegistration.value(msg.value)(_id);
+    }
+
+
+    /** @dev Put proposal to support voting only when a new proposal is permitted.
+     *  @param _id ID of a proposal
+     */
+    function putProposalToSupport(bytes32 _id) public onlyWhenProposalIsNew(_id) onlyWhenPermitted(_id) {
+        Proposal storage proposal = proposals[_id];
+
+        address cloneToken = pinakion.createCloneToken({_cloneTokenName: quorumTokenName, _cloneDecimalUnits: DECIMALS, _cloneTokenSymbol: quorumTokenSymbol, _snapshotBlock: block.number, _transfersEnabled: true});
+        proposal.quorumToken = MiniMeToken(cloneToken);
+        proposal.quorumToken.changeController(tokenController);
+
+        proposal.state = ProposalState.PutToSupport;
+
+        emit ProposalPutToSupport(_id);
+    }
+
+
+    /** @dev Calculate and return required quorum for a given proposal.
+     *  @param _id ID of a proposal
+     */
+    function getRequiredQuorum(bytes32 _id) public view onlyWhenProposalPutToSupport(_id) returns (uint effectiveQuorum){
+        uint numberOfDividePeriodsPassed = (block.timestamp - lastTimeQuorumReached) / quorumDivideTime;
+        effectiveQuorum = currentProposalQuorum * proposals[_id].quorumToken.totalSupply() / (2 ** numberOfDividePeriodsPassed) / 100;
+    }
+
+
+    /** @dev Put given proposal to vote.
+     *  @param _id ID of a proposal
+     */
+    function putProposalToVote(bytes32 _id) public onlyWhenProposalPutToSupport(_id) onlyWhenQuorumReached(_id) {
+        Proposal storage proposal = proposals[_id];
+
+        proposal.whenPutToVote = block.timestamp;
+
+        address cloneToken = pinakion.createCloneToken({_cloneTokenName: voteTokenName, _cloneDecimalUnits: DECIMALS, _cloneTokenSymbol: voteTokenSymbol, _snapshotBlock: block.number, _transfersEnabled: true});
+        proposal.voteToken = MiniMeToken(cloneToken);
+        proposal.voteToken.changeController(tokenController);
+
+        proposal.state = ProposalState.PutToVote;
+
+        emit ProposalPutToVote(_id);
+
+        resetSettings(); // Reset global variables as quorum has reached
+    }
+
+
+    /** @dev Ends a voting, moves proposal to decided state, sets the decision
+     *  @param _id ID of a proposal
+     */
+    function finalizeVoting(bytes32 _id) public onlyWhenProposalPutToVote(_id) onlyWhenVotingPeriodPassed(_id) {
```

Comment:
The two modifiers are redundant, you might have other cases like these in other functions too.